x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/utils.ts

```diff
@@ -26,47 +21,26 @@ export const ruleStepsOrder: RuleStepsOrder = [
   RuleStep.ruleActions,
 ];
 
-const getRulesBreadcrumb = (pathname: string, search: string[], getUrlForApp: GetUrlForApp) => {
-  const tabPath = pathname.split('/')[1];
-
-  if (tabPath === 'rules') {
-    return {
-      text: i18nRules.PAGE_TITLE,
-      href: getUrlForApp(APP_UI_ID, {
-        deepLinkId: SecurityPageName.rules,
-        path: getRulesUrl(!isEmpty(search[0]) ? search[0] : ''),
-      }),
-    };
-  }
-};
-
 const isRuleCreatePage = (pathname: string) =>
   pathname.includes(RULES_PATH) && pathname.includes('/create');
 
 const isRuleEditPage = (pathname: string) =>
   pathname.includes(RULES_PATH) && pathname.includes('/edit');
 
-export const getBreadcrumbs = (
+export const getTrailingBreadcrumbs = (
```

Comment:
Much nicer name! 🔥  I am quite sure you also noticed that this logic si copied in several places. Perhaps we can extract it into a hook?

Reply:
We initially thought about that, but we discovered that all sub plugins have some peculiarity.
The code that generates the path before the plugin was extracted to a shared place. So now, what is left are the inner plugins breadcrumbs, and those can be different for every plugin. 

`getTrailingBreadcrumbs` peculiarities:
* The management sub plugin has a map `TabNameMappedToI18nKey` for the title, and every page inside is identified by the'tabName`.
* The network sub plugin has to decode `Ipv6`
* The Rule page is the only one with `RuleCreatePage` and `RuleEditPage` breadcrumbs.
* User and Host have the same tab structure because we created the first version of the User page as a copy of the Host page. But it should change soon.

We decided to allow every sub plugin to implement its version of `getTrailingBreadcrumbs` instead of creating a function that tries to deal with every edge case. 

Maybe in the feature, if those pages don't change. Then, we could create a reusable helper for Host, Network, and User pages. Those look like the most similar ones because they all have tabs. 

@ashokaditya please, let me know if you agree with this decision.

Reply:
I see what you mean. Thanks for taking the time to explain @machadoum.